src/Microsoft.AspNet.OData.Shared/ODataQueryContextExtensions.cs

```diff
@@ -26,5 +27,23 @@ public static ODataQuerySettings UpdateQuerySettings(this ODataQueryContext cont
 
             return updatedSettings;
         }
+
+        public static SkipTokenHandler GetSkipTokenHandler(this ODataQueryContext context)
+        {
+            if (context == null || context.RequestContainer == null)
+            {
+                return DefaultSkipTokenHandler.Instance;
+            }
+            return context.RequestContainer.GetRequiredService<SkipTokenHandler>();
+        }
+
+        public static SkipTokenQueryValidator GetSkipTokenQueryValidator(this ODataQueryContext context)
+        {
+            if (context == null || context.RequestContainer == null)
+            {
+                return new SkipTokenQueryValidator();
```

Comment:
>new SkipTokenQueryValidator() [](start = 23, length = 29)

can we use the default instance here also?

Reply:
[https://github.com/OData/WebApi/issues/1805](https://github.com/OData/WebApi/issues/1805)

---
In reply to: [283532482](https://github.com/OData/WebApi/pull/1724#discussion_r283532482) [](ancestors = 283532482)